### PR TITLE
Server update db logic

### PR DESCRIPTION
**Pull request by stephen-lt-sl**
@stevenhillcox @sievins @jebbinSCL

Updates the DB-interface logic on the server to use the current version of the API, including replacing all callbacks with promises.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -114,36 +113,29 @@ module.exports = function(port, db, githubAuthoriser) {
...
-                        });
-                    } else {
-                        res.sendStatus(500);
-                    }
-                });
-            } else {
-                res.sendStatus(500);
+        }).limit(1).next().then(function(sender) {
+            if (!sender) {
+                throw false;
```
Comment (jebbinSCL) [inline server/server.js]: I'm not sure about this, so it would be good for you to look this up. Should you be throwing here, or should you rejecting the promise? I expected a reject, but the behaviour of them is a little different, so make sure this is correct.

Comment (jebbinSCL) [inline server/server.js]: From a quick search, there might be some edge cases where the throw doesn't work as expected. Take a look at let me know what you find?

**Comment by jebbinSCL** [comment]
As a part of your refactoring, I think its time to start treating your server tests as normal code and see whether there is a clean way to reduce the duplication.

File: server/server.js (hunk 1 above)
Comment (stephen-lt-sl) [reply server/server.js]: I think you're right about this, I've replaced the calls to throw with rejected promises - it seems cleaner, more consistent with the surrounding code, and more functionally accurate for what the code is supposed to do.

Comment (jebbinSCL) [inline server/server.js]: Cool, good to know for the future.

**Comment by stephen-lt-sl** [comment]
As suggested I've tried moving functionality from the server tests to a separate helper module; as part of the transition the tests were updated to make better use of promises, particularly in relation to Mocha. As a result the line-change count is very high, even though functionally the system behaves the same as before.

It would be possible to decompose this into several smaller sequential pull requests, but they would each have comparably high line-change counts, as they all change roughly the same patches of code (the best bet is probably to split the initial refactoring, up to "Correctly return rejected promises," from the additional refactoring).

I also highly recommend using the 'Split' view option when looking at these changes, as the 'Unified' view is incredibly convoluted in comparison.

**Comment by jebbinSCL** [comment]
:ship: for now. In a separate pull request, you should be able to split up the files for the tests as well?
